gtsam/hybrid/HybridFactorGraph.h

```diff
@@ -0,0 +1,110 @@
+/* ----------------------------------------------------------------------------
+
+ * GTSAM Copyright 2010, Georgia Tech Research Corporation,
+ * Atlanta, Georgia 30332-0415
+ * All Rights Reserved
+ * Authors: Frank Dellaert, et al. (see THANKS for the full author list)
+
+ * See LICENSE for the license information
+
+ * -------------------------------------------------------------------------- */
+
+/**
+ * @file   HybridFactorGraph.h
+ * @brief  Hybrid factor graph that uses type erasure
+ * @author Fan Jiang
+ * @date   Mar 11, 2022
+ */
+
+#pragma once
+
+#include <gtsam/hybrid/HybridFactor.h>
+#include <gtsam/inference/EliminateableFactorGraph.h>
+#include <gtsam/inference/FactorGraph.h>
+#include <gtsam/inference/Ordering.h>
+
+namespace gtsam {
+
+// Forward declarations
+class HybridFactorGraph;
+class HybridConditional;
+class HybridBayesNet;
+class HybridEliminationTree;
+class HybridBayesTree;
+class HybridJunctionTree;
+class DecisionTreeFactor;
+
+class JacobianFactor;
+
+/** Main elimination function for HybridFactorGraph */
+GTSAM_EXPORT
+std::pair<boost::shared_ptr<HybridConditional>, HybridFactor::shared_ptr>
+EliminateHybrid(const HybridFactorGraph& factors, const Ordering& keys);
+
+/* ************************************************************************* */
+template <>
+struct EliminationTraits<HybridFactorGraph> {
+  typedef HybridFactor FactorType;  ///< Type of factors in factor graph
+  typedef HybridFactorGraph
+      FactorGraphType;  ///< Type of the factor graph (e.g. HybridFactorGraph)
+  typedef HybridConditional
+      ConditionalType;  ///< Type of conditionals from elimination
+  typedef HybridBayesNet
+      BayesNetType;  ///< Type of Bayes net from sequential elimination
+  typedef HybridEliminationTree
+      EliminationTreeType;                      ///< Type of elimination tree
+  typedef HybridBayesTree BayesTreeType;        ///< Type of Bayes tree
+  typedef HybridJunctionTree JunctionTreeType;  ///< Type of Junction tree
+  /// The default dense elimination function
+  static std::pair<boost::shared_ptr<ConditionalType>,
+                   boost::shared_ptr<FactorType> >
+  DefaultEliminate(const FactorGraphType& factors, const Ordering& keys) {
+    return EliminateHybrid(factors, keys);
+  }
+};
+
+/**
+ * Hybrid Factor Graph
+ * -----------------------
+ * This is the linear version of a hybrid factor graph. Everything inside needs
```

Comment:
What does "linear" mean in this context? Consider re-phrasing.

Reply:
I think this may not be right. We don't have a nonlinear/linear dichotomy for the Hybrid Factor Graph. Seems like `HybridFactorGraph` should be renamed to `GaussianHybridFactorGraph`.

Reply:
We used `GaussianHybridFactorGraph` in the previous code. It made sense since we have `HybridFactorGraph - GaussianHybridFactorGraph - NonlinearHybridFactorGraph`. I am not sure why @ProfFan set the name to `HybridGaussianFactor`, so what should we do?

Reply:
I think this can be renamed to GaussianHybridFactorGraph